vm/src/stdlib/marshal.rs

```diff
@@ -30,25 +78,108 @@ mod decl {
         Ok(())
     }
 
+    /// Read the next 8 bytes of a slice, convert to usize
+    /// Has side effect of increasing position pointer
+    fn eat_usize(bytes: &[u8], position: &mut usize) -> usize {
```

Comment:
Errors could be stemming from this function. Not all architectures have a usize of 8 bytes (could be 2, 4, 8). So instead of reading usize, I am going to switch marshaling to use u32 despite architecture, then convert back to usize. So the only possible error is an element or list being longer than 2^32 bytes, in which case we send an error that it can't be marshaled